rustls/tests/api.rs

```diff
@@ -3345,8 +3345,8 @@ mod test_quic {
             }
 
             let (x, y) = (keys(x), keys(y));
-            equal_packet_keys(&x.local.packet, &y.remote.packet)
-                && equal_packet_keys(&x.remote.packet, &y.local.packet)
+            equal_packet_keys(&*x.local.packet, &*y.remote.packet)
+                && equal_packet_keys(&*x.remote.packet, &*y.local.packet)
```

Comment:
I think this might be a style thing, but personally I think `as_ref()` reads better than `&*`:
```rust
            equal_packet_keys(x.local.packet.as_ref(), y.remote.packet.as_ref())
                && equal_packet_keys(x.remote.packet.as_ref(), y.local.packet.as_ref())
```
WDYT?

I think there are some other `&*` instances in this general area that don't appear to be required, e.g. L3328 & L3332 can just be `header` instead of `&*header`. Am I missing something deeper here?

Reply:
have gone back to `.as_ref()` here and elsewhere